api-reference/10 UI Components/dxAutocomplete/1 Configuration/valueExpr.md

```diff
@@ -1,3 +1,5 @@
-Always set this property if the [dataSource](/api-reference/10%20UI%20Widgets/DataExpressionMixin/1%20Configuration/dataSource.md '{basewidgetpath}/Configuration/#dataSource') contains objects. Alternatively, you can specify the [key](/api-reference/30%20Data%20Layer/Store/1%20Configuration/key.md '/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#key') in the store. 
+When [dataSource]({basewidgetpath}/Configuration/#dataSource) contains objects, we encourage you to define **valueExpr** to correctly identify data items. Otherwise, the component uses referential equality to compare objects and may result in unexpected behavior.
```

Comment:
When [dataSource]({basewidgetpath}/Configuration/#dataSource) contains objects, you should define **valueExpr** to correctly identify data items. Otherwise, the component uses referential equality to compare objects that may result in unexpected behavior.